examples/user_interfaces/embed_in_paraview_sgskip.py

```diff
@@ -0,0 +1,97 @@
+"""
+===================================
+Embedding in a Paraview application
+===================================
+
+
+`ParaView <https://www.paraview.org/>`__ is a powerful 3D
+visualization tool.  It has both a Qt client application and python
+integration.
+
+This example shows how to embed an interactive Matplotlib figure into
+the Qt UI.
+
+ParaView (via VTK) embeds a python interpreter into the primary
+application.  This code is intended to be imported into the
+interpreter running within ParaView, find the main window and
+inject a dock widget with a Matplotlib Figure in it.  To use this,
+in the ParaView python shell ::
+
+ >> import sys
+ >> sys.path.append(path_to_examples)
+ >> import embed_in_paraview_sgksip as eip
+ >> fig = eip.make_dock_figure()
+ >> ax = fig.add_subplot(111)  # or ax = fig.subplots()
+ >> ln, = ax.plot(range(5))
+
+"""
+
+
+from PyQt5 import QtWidgets, QtCore
+from matplotlib.figure import Figure
+from matplotlib.backends.backend_qt5agg import (
+    FigureCanvas, NavigationToolbar2QT)
+
+
+def get_main_window(target_name='pqClientMainWindow'):
+    '''Find the main window
+
+    This assumes that the object name on the main window (where we
+    will be able to add a dock widget) has the object name
+
+    Parameters
+    ----------
+    target_name : str
+        The same of the MainWindow object of interest
+
+    Returns
+    -------
+    QtWidgets.QMainWindow
+
+    '''
+    app = QtWidgets.QApplication.instance()
+    if app is None:
+        raise RuntimeError("No running QApplication found. "
+                           "May not be running inside of ParaView")
+
+    return next(iter(
+        w for w in app.topLevelWidgets()
```

Comment:
app.toplevelwidgets() if app else []

and catch stopiteration

should *actually* cover the case where paraview is not running?

Reply:
It is intended to blow up if you are not inside of paraview.

The first case should never be hit due to the `if` block above.

Reply:
My point is that if you already did the work to blow up with a nice error message in half of the cases, you may as well do so in all cases (remove the none check above and throw the exception below)